### PR TITLE
Fixes template objects describer

### PR DESCRIPTION
**Pull request by fabianofranz**
Fixes https://github.com/openshift/origin/issues/14058 (template describer)

Avoid the universal decoder (use only `UnstructuredJSONScheme`) when decoding the list of template objects to bypass validation in fields like "quantity". Read `Kind` from the unstructured content.

PTAL @juanvallejo @deads2k @ncdc

### CODE REVIEW

**Comment by fabianofranz** [comment]
[test]

File: pkg/cmd/cli/describe/describer.go [hunk 1]
```diff
@@ -1042,6 +1043,15 @@ func (d *TemplateDescriber) describeObjects(objects []runtime.Object, out *tabwr
 		meta.Name, _ = d.MetadataAccessor.Name(obj)
 		gvk, _, err := d.ObjectTyper.ObjectKinds(obj)
 		if err != nil {
+			if unstructured, ok := obj.(runtime.Unstructured); ok {
+				content := unstructured.UnstructuredContent()
```
Comment (deads2k) [inline pkg/cmd/cli/describe/describer.go]: `unstructured.GetObjectKind()`?

**Comment by deads2k** [comment]
I think we should try using our meta functions instead of direct access.

File: pkg/cmd/cli/describe/describer.go [hunk 2]
```diff
@@ -1042,6 +1043,15 @@ func (d *TemplateDescriber) describeObjects(objects []runtime.Object, out *tabwr
...
+				if objKind, ok := content["kind"]; ok {
+					if objKindStr, ok := objKind.(string); ok {
+						fmt.Fprintf(out, fmt.Sprintf("%s%s\t%s\n", indent, objKindStr, meta.Name))
```
Comment (deads2k) [inline pkg/cmd/cli/describe/describer.go]: `kind.group` I think.

**Comment by fabianofranz** [comment]
@deads2k improved to use meta, ptal.

File: pkg/cmd/cli/describe/describer.go [hunk 3]
```diff
@@ -1038,14 +1039,20 @@ func (d *TemplateDescriber) describeObjects(objects []runtime.Object, out *tabwr
...
+		if err != nil {
+			gvk, _, err := d.ObjectTyper.ObjectKinds(obj)
+			if err != nil {
+				kind = "<unknown>"
+			} else {
+				kind = gvk[0].Kind
+			}
 		}
-		fmt.Fprintf(out, fmt.Sprintf("%s%s\t%s\n", indent, gvk[0].Kind, meta.Name))
+		fmt.Fprintf(out, fmt.Sprintf("%s%s\t%s\n", indent, kind, name))
```
Comment (deads2k) [inline pkg/cmd/cli/describe/describer.go]: You need group information for disambiguation

**Comment by fabianofranz** [comment]
> You need group information for disambiguation

Good catch, and looks like it had the issue already. Updated, @deads2k ptal?

**Comment by deads2k** [comment]
lgtm [merge]

**Pull request merged**